### PR TITLE
assertrepr_compare: use safeformat with -vv

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/pytest-dev/pytest/issues/3962

TODO:

- [x] changelog

### CODE REVIEW

File: src/_pytest/assertion/util.py [hunk 1]
```diff
@@ -123,13 +124,21 @@ def isiterable(obj):
 
 def assertrepr_compare(config, op, left, right):
     """Return specialised explanations for some operators/operands"""
-    maxsize = (80 - 15 - len(op) - 2) // 2  # 15 chars indentation, 1 space around op
-    left_repr = saferepr(left, maxsize=maxsize)
-    right_repr = saferepr(right, maxsize=maxsize)
+    verbose = config.getoption("verbose")
+    if verbose > 1:
+        left_repr = safeformat(left)
```
Comment (nicoddemus) [inline src/_pytest/assertion/util.py]: So the idea here is to don't use `maxsize` with `-vv`? Seems reasonable.

Please add a CHANGELOG entry as well.

**Comment by blueyed** [comment]
Uncovered a bug in pytest-xdist: https://github.com/pytest-dev/pytest-xdist/pull/478

**Comment by blueyed** [comment]
Only added changelog, should be good.

**Comment by blueyed** [comment]
Should I pull out the CI improvement into a separate PR, or can we merge this already?

**Pull request merged**